packages/tasit-action/src/TasitAction.test.js

```diff
@@ -265,81 +354,164 @@ describe("TasitAction.Contract", () => {
+          const { args } = data;
+
+          fakeFn();
+          resolve();
+        });
+      });
+
+      contractSubscription.off("error");
+
+      expect(errorFakeFn.called).to.be.false;
+      expect(fakeFn.called).to.be.true;
+    });
+
+    it("should throw error when listening on invalid event", async () => {
+      contractSubscription = simpleStorage.subscribe();
 
       expect(() => {
-        subscription.on("InvalidEvent", () => {});
+        contractSubscription.on("InvalidEvent", () => {});
       }).to.throw();
     });
 
-    // FIXME: Non-deterministic tests - Quarantine
-    // More info: https://github.com/tasitlabs/TasitSDK/pull/95
-    it.skip("should listen to an event", async () => {
-      subscription = simpleStorage.subscribe();
-      const fakeFn = sinon.fake();
+    it("subscription should has one listener per event", async () => {
+      contractSubscription = simpleStorage.subscribe();
 
-      const handlerFunction = async message => {
-        const { data } = message;
-        const { args } = data;
-        fakeFn();
+      const listener1 = message => {
+        console.log("1");
+      };
+      const listener2 = message => {
+        console.log("2");
       };
 
-      await subscription.on("ValueChanged", handlerFunction);
+      expect(contractSubscription.eventNames()).to.be.empty;
 
-      simpleStorage.setValue("hello world");
+      contractSubscription.on("ValueChanged", listener1);
 
-      await mineBlocks(provider, 15);
+      expect(() => {
+        contractSubscription.on("ValueChanged", listener2);
+      }).to.throw();
 
-      expect(fakeFn.called).to.be.true;
+      expect(contractSubscription.eventNames()).to.deep.equal(["ValueChanged"]);
+    });
+
+    it("should remove an event", async () => {
+      contractSubscription = simpleStorage.subscribe();
+
+      const listener1 = message => {};
+
+      expect(contractSubscription.eventNames()).to.be.empty;
+
+      contractSubscription.on("ValueChanged", listener1);
+
+      expect(contractSubscription.eventNames()).to.deep.equal(["ValueChanged"]);
+
+      contractSubscription.off("ValueChanged");
+
+      expect(contractSubscription.eventNames()).to.be.empty;
     });
 
     it("should manage many listeners", async () => {
-      subscription = simpleStorage.subscribe();
+      contractSubscription = simpleStorage.subscribe();
 
       const listener1 = message => {};
       const listener2 = message => {};
       const listener3 = message => {};
 
-      expect(subscription.listenerCount("ValueChanged")).to.equal(0);
+      expect(contractSubscription.eventNames()).to.be.empty;
 
-      subscription.on("ValueChanged", listener1);
-      subscription.on("ValueChanged", listener2);
+      contractSubscription.on("ValueChanged", listener1);
+      contractSubscription.on("ValueRemoved", listener2);
 
-      expect(subscription.listenerCount("ValueChanged")).to.equal(2);
+      expect(contractSubscription.eventNames()).to.deep.equal([
```

Comment:
Does the order matter?

Reply:
Sorry, which order exactly?

Reply:
The order of the event-names in the array (which is shows up in the next line in the code). I was wondering if we return them in the same order AND whether deep equal cares what the order is.

Reply:
Okay, thanks! Resolving this.